e2e/cephfs.go

```diff
@@ -126,14 +129,53 @@ var _ = Describe("cephfs", func() {
 					Fail(err.Error())
 				}
 			}
+			if upgradeTesting {
+				// fetch current working directory to switch back
+				// when we are done upgrading.
+				cwd, err := os.Getwd()
+				if err != nil {
+					Fail(err.Error())
+				}
+				err = upgradeAndDeployCSI(upgradeVersion, "cephfs")
+				if err != nil {
+					Fail(err.Error())
+				}
+				createConfigMap(cephfsDirPath, f.ClientSet, f)
+				createCephfsSecret(f.ClientSet, f)
+				createCephfsStorageClass(f.ClientSet, f, true, "")
+				pvcPath := cephfsExamplePath + "pvc.yaml"
+				appPath := cephfsExamplePath + "pod.yaml"
+				_, app := createPVCAndAppBinding(pvcPath, appPath, f, deployTimeout)
+				err = deletePod(app.Name, app.Namespace, f.ClientSet, deployTimeout)
+				if err != nil {
+					Fail(err.Error())
+				}
+				deleteCephfsPlugin()
+
+				// support of PSP is available in current master, so we don't need to
+				// skip it now.
+				skipPSP = false
+				err = os.Chdir(cwd)
+				if err != nil {
+					Fail(err.Error())
+				}
+				deployCephfsPlugin()
+				// validate if the app gets bound to a pvc created by
+				// an earlier release.
+				err = createApp(f.ClientSet, app, deployTimeout)
```

Comment:
move this to a new `By()` as we discussed earlier you also need to delete PVC and verify it.

Reply:
Since we have created separate jobs in e2e for upgrade testing, I checked it before the e2e tests and skip the rest.... Correct me if I am wrong, but including it in `By()` means we will have to start running e2e tests which we are skipping for upgrade testing job to avoid redundant checks.... Any thoughts?

Reply:
lets create a separate file for upgrade testing with describe , in before each handle deploy from release version and we can have many By function to do extra functionalities

Reply:
Sounds good. We can do that. Thanks.